README.md

```diff
@@ -1,23 +1,62 @@
 # coffee-caller
 
+## Architecture
+
+```mermaid
+    C4Context
+      title System Context diagram for CoffeeCaller
```

Comment:
Minor: The whole system already has the title CoffeeCaller, I dont need the name within the sub modules as well. 
Suggestion: CoffeeCallerDesktopFrontend -> DesktopFrontend, etc.